src/main/java/vazkii/patchouli/common/base/PatchouliAPIImpl.java

```diff
@@ -104,6 +108,18 @@ public ITextComponent getSubtitle(@Nonnull ResourceLocation bookId) {
 		return book.getSubtitle();
 	}
 
+	@Override
+	@OnlyIn(Dist.CLIENT)
+	public void registerCommand(String name, Function<IStyleStack, String> command) {
+		BookTextParser.register(style -> command.apply(style), name);
+	}
+
+	@Override
+	@OnlyIn(Dist.CLIENT)
+	public void registerFunction(String name, BiFunction<String, IStyleStack, String> function) {
+		BookTextParser.register((arg, style) -> function.apply(arg, style), name);
```

Comment:
any reason we can't get rid of the internal interfaces?

also, it'd be cool if Patchouli could register through the api just to exercise the call path.

Reply:
well, the issue is that some internal stuff actually _uses_ SpanState and not just what's exposed in IStyleStack, so the registrations do in fact have to use the "upper" interfaces